core/components/Behaviour.lua

```diff
@@ -0,0 +1,10 @@
+local Behaviour = {}
+Behaviour.__index = Behaviour
+Behaviour.new = function(onEnter, onUpdate, onExit)
+  local self = setmetatable({},Behaviour)
+  self.onEnter = onEnter or nil
+  self.onExit = onExit or nil
+  self.onUpdate = onUpdate or nil
```

Comment:
Behaviour was a *very* convenient place to hack in non-state-change behaviours. It might be worth building out what this class does.